### PR TITLE
solved task 10 shopping

### PR DESCRIPTION
**Pull request by neb5562**

### CODE REVIEW

File: 10 - shopping/solution_10.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def initialize
     @shopping_total = 0.00
   end
   
-  def start
+  def get_user_input
```
Comment (neb5562) [inline 10 - shopping/solution_10.rb:11]: changed method name `start `to `get_user_input`

File: 10 - shopping/solution_10.rb [hunk 2]
```diff
@@ -33,18 +33,31 @@ def store_item product_name, product_price, product_quantity
 
   # display final shopping summary
   def display_result
-    puts "\nShopping Summary:\n"
-    puts "-----------------\n"
     item_number = 1
+    print_header
     @items.each do | item_key, item_value |
-      puts "##{item_number} - #{item_key} Quantity: #{item_value["product_quantity"]}, Item Price: #{item_value["product_price"]}, Item Total Price: #{ (item_value["product_quantity"].to_f * item_value["product_price"].to_f).round(PRICE_ROUND_PRECISION) }\n"
```
Comment (neb5562) [inline 10 - shopping/solution_10.rb]: moved prints in small methods

File: 10 - shopping/solution_10.rb [hunk 3]
```diff
@@ -35,10 +35,14 @@ def store_item product_name, product_price, product_quantity
   def display_result
     item_number = 1
     print_header
-    @items.each do | item_key, item_value |
-      print_item item_number, item_key, item_value
-      item_number += 1
-      @shopping_total += (item_value["product_quantity"] * item_value["product_price"])
+    if !@items.empty?
```
Comment (neb5562) [inline 10 - shopping/solution_10.rb:38]: if hash variable is empty, print result

File: 10 - shopping/solution_10.rb [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  end
+
+  private
+  def print_header
+    puts "\nShopping Summary:\n"
+    puts "-----------------\n"
+  end
+
+  def print_item item_number, item_key, item_value
+    puts "##{item_number} - #{item_key} | Quantity: #{item_value["product_quantity"]} | Item Price: #{item_value["product_price"]} | Item Total Price: #{ (item_value["product_quantity"].to_f * item_value["product_price"].to_f).round(PRICE_ROUND_PRECISION) }\n"
```
Comment (alexshvedko) [inline 10 - shopping/solution_10.rb]: long line

File: 10 - shopping/solution_10.rb [hunk 5]
```diff
@@ -8,9 +8,9 @@ def initialize
     @shopping_total = 0.00
   end
   
-  def start
+  def get_user_input
     while true
-      puts "Product Name:"
+      puts "Product Name: (please enter 'stop' to end shopping.)"
```
Comment (neb5562) [inline 10 - shopping/solution_10.rb:13]: added information to user how to stop shopping

File: 10 - shopping/solution_10.rb (hunk 4 above)
Comment (neb5562) [reply 10 - shopping/solution_10.rb]: > long line

fixed

**Review by alexshvedko — APPROVED** [review]

**Pull request merged**